docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,479 @@
+
+Since Rust doesn't have inheritance, it's not possible to inherit from `Node` as is common practice in [`rclcpp`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html) or [`rclpy`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html).
+
+The code used in these examples can be found [here](https://github.com/ros2-rust/ros2_rust/tree/main/examples/rust_pubsub)
+<div style="margin-left:20px;">
+<details><summary>Side-note on dependencies</summary>
+
+You may be wondering why you can't just add all your ROS 2-specific dependencies to `Cargo.toml` with `cargo add YOUR_DEPENDENCIES` and have to edit this file manually. Here is why:
+Almost none of the ROS 2 dependencies you'll need for your ROS 2 Rust node development currently exist on [crates.io](https://crates.io/), the 
+main source for Rust dependencies. So the add command simply can't find the dependency targets. What colcon does by compiling 
+the ROS 2 Rust dependencies and your ROS 2 Rust project is redirect the cargo search for dependencies directly into your 
+`workspace/install` folder, where it'll find locally generated Rust projects to use as dependencies. In particular, almost 
+all message types will be called as dependencies for your ROS 2 Rust project this way.
+
+</details></div>
+
+</details>
+
+<details><summary>Prerequisites </summary> 
+
+Basic concepts of development with ROS 2 should be known:
+* [workspaces](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Creating-A-Workspace/Creating-A-Workspace.html)
+* [packages](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html).
+
+A basic understanding of [Rust](https://doc.rust-lang.org/book/) is recommended, but not entirely necessary.
+Before developing [ros2-rust](https://github.com/ros2-rust/ros2_rust) nodes, you must follow the 
+[installation instructions](https://github.com/ros2-rust/ros2-rust/blob/main/README.md) for [`rclrs`](https://docs.rs/rclrs/latest/rclrs/).
+For a full Introduction into Rust, please read the very good [Rust book](https://doc.rust-lang.org/book/title-page.html).
+
+</details>
+
+<details><summary>Tasks </summary> 
+<div style="margin-left:20px;"><details><summary>Create a Package</summary>
+
+Currently, building a package for ros2-rust is different 
+from building packages for [Python](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html) or [C/C++](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html).  
+
+First, you'll need to create a standard [cargo](https://doc.rust-lang.org/cargo/) 
+package:
+```
```

Comment:
```suggestion
```console
```

Reply:
addressed

Reply:
I didn't knew that.

Reply:
That was too inconsistent for me. I have replaced each `shell` and `console` hint with an `sh`. so it is consistent for normal terminal calls.